clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/InfrastructureIT.java

```diff
@@ -838,4 +838,484 @@ public void shouldGetRawResources() throws InterruptedException, IOException {
+            + ACCOUNT1_NAME
+            + "/kubernetes"
+            + " to return valid data");
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given two kubernetes network policies for one account\n"
+          + "When sending get securityGroups/{account}/{cloudprovider}/{name}\n"
+          + "Then response should contain the security group specified in name\n===")
+  @Test
+  public void shouldListSecurityGroupsByAccountAndCloudProviderAndName()
+      throws InterruptedException, IOException {
+    // ------------------------- given --------------------------
+    String ns = kubeCluster.getAvailableNamespace();
+    String name = "default-deny-ingress";
+    String appName = "getmanifest";
+    System.out.println("> Using namespace " + ns);
+
+    Map<String, Object> networkPolicy =
+        KubeTestUtils.loadYaml("classpath:manifests/networkpolicy.yml")
+            .withValue("metadata.namespace", ns)
+            .withValue("metadata.name", name)
+            .asMap();
+
+    List<Map<String, Object>> body =
+        KubeTestUtils.loadJson("classpath:requests/deploy_manifest.json")
+            .withValue("deployManifest.account", ACCOUNT1_NAME)
+            .withValue("deployManifest.moniker.app", appName)
+            .withValue("deployManifest.manifests", List.of(networkPolicy))
+            .asList();
+
+    KubeTestUtils.deployAndWaitStable(baseUrl(), body, ns, "networkPolicy" + " " + name);
+
+    Map<String, Object> networkPolicy2 =
+        KubeTestUtils.loadYaml("classpath:manifests/networkpolicy.yml")
+            .withValue("metadata.namespace", ns)
+            .withValue("metadata.name", name + "-second")
+            .asMap();
+
+    List<Map<String, Object>> body2 =
+        KubeTestUtils.loadJson("classpath:requests/deploy_manifest.json")
+            .withValue("deployManifest.account", ACCOUNT1_NAME)
+            .withValue("deployManifest.moniker.app", appName)
+            .withValue("deployManifest.manifests", List.of(networkPolicy2))
+            .asList();
+
+    KubeTestUtils.deployAndWaitStable(
+        baseUrl(), body2, ns, "networkPolicy" + " " + name + "-second");
+
+    KubeTestUtils.repeatUntilTrue(
+        () -> {
+          // ------------------------- when --------------------------
+          System.out.println("> Sending get securityGroups request");
+          Response resp =
+              get(
+                  baseUrl()
+                      + "/securityGroups/"
+                      + ACCOUNT1_NAME
+                      + "/kubernetes/networkpolicy "
+                      + name);
+
+          // ------------------------- then --------------------------
+          System.out.println(resp.asString());
+          if (resp.statusCode() == 404) {
+            return false;
+          }
+          resp.then().statusCode(200);
+          List securityGroupList = resp.jsonPath().getList(ns);
+          return securityGroupList != null && securityGroupList.size() > 0;
```

Comment:
Since you're querying for name, this list should return exactly one item right? Otherwise you wouldn't discover bugs where the response doesn't filter properly by name.